Sage/carbon
# Sidebar Header and Form Footer Padding FE-5391

### Proposed behaviour


Adds `footerPadding` prop to support padding interface on Form footer component.
Updates `form-style-util` to ensure any padding overrides applied to footer when
a child of `Sidebar` are only applied when no value is passed to `footerPadding`
prop

Surfaces `headerPadding` prop to support the ability to override the padding in the `Sidebar`
`header`

![image](https://user-images.githubusercontent.com/44157880/203048974-84d4b526-bf08-4631-94fa-bb4b395708e0.png)

### Current behaviour


No support for custom padding in `Sidebar` header or `Form` footer

### Checklist




#### QA


### Additional context



### Testing instructions


See linked sandbox and added stories, check that when Form footer has padding the default behaviour is overridden (no padding is applied by the `form-style-util`)

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


https://codesandbox.io/s/tender-sound-7blp6k?file=/src/index.js:1122-1123
